fabric_cf/orchestrator/core/orchestrator_slice_wrapper.py

```diff
@@ -163,6 +227,117 @@ def create(self, *, bqm_graph: Neo4jCBMGraph, slice_graph: Neo4jASM,
+            elif sliver_type == ServiceType.L2STS or sliver_type == ServiceType.L2Bridge or \
+                    sliver_type == ServiceType.L2PTP:
+
+                self.logger.trace(f"Network Service Sliver: {sliver}")
+
+                # Processing Interface Slivers
+                if sliver.interface_info is not None:
+                    predecessor_reservations = []
+                    for ifs in sliver.interface_info.interfaces.values():
+                        # Get Mapping information for Interface Sliver from ASM
+                        # i.e. [Peer IFS, Peer NS Id, Component Name, Node Id]
+                        ifs_mapping = FimHelper.get_interface_sliver_mapping(ifs_node_id=ifs.node_id,
+                                                                             slice_graph=slice_graph)
+
+                        if ifs_mapping is None:
+                            raise OrchestratorException(message=f"Peer connection point not found for ifs# {ifs}",
+                                                        http_error_code=OrchestratorException.HTTP_BAD_REQUEST)
+
+                        # capacities (bw in Gbps, burst size is in Mbits) source: (b)
+                        # Set Capacities
+                        ifs.set_capacities(cap=ifs_mapping.get_peer_ifs().get_capacities())
+                        # Set Labels
+                        ifs.set_labels(lab=ifs_mapping.get_peer_ifs().get_labels())
+
+                        # Save the parent component name and the parent reservation id in the Node Map
+                        parent_res_id = node_res_mapping.get(ifs_mapping.get_node_id(), None)
+
+                        node_map = tuple([ifs_mapping.get_component_name(), parent_res_id])
+                        ifs.set_node_map(node_map=node_map)
+
+                        self.logger.trace(f"Interface Sliver: {ifs}")
+
+                        if parent_res_id is not None and parent_res_id not in predecessor_reservations:
+                            predecessor_reservations.append(parent_res_id)
+
+                    # Generate reservation for the sliver
+                    reservation = self.reservation_converter.generate_reservation(sliver=sliver,
+                                                                                  slice_id=self.slice_obj.get_slice_id(),
+                                                                                  end_time=self.slice_obj.get_lease_end(),
+                                                                                  pred_list=predecessor_reservations)
+                    reservations.append(reservation)
+            else:
+
+                raise OrchestratorException(message="Not implemented",
+                                            http_error_code=OrchestratorException.HTTP_BAD_REQUEST)
+        return reservations
+
+    def __build_network_node_reservations(self, slice_graph: Neo4jASM) -> Tuple[List[TicketReservationAvro], Dict[str, str]]:
```

Comment:
This is minor - whenever possible it is better to use the abstract versions (ABCASM, instead of concrete) especially in type designators.